loadgen/adapters/sidecar_receiver.go

```diff
@@ -74,29 +93,7 @@ func receiveCommittedBlock(
 			if !ok {
 				return
 			}
-
-			statusCodes := block.Metadata.Metadata[common.BlockMetadataIndex_TRANSACTIONS_FILTER]
-			logger.Infof("Received block #%d with %d TXs and %d statuses [%s]",
-				block.Header.Number, len(block.Data.Data), len(statusCodes), recapStatusCodes(statusCodes),
-			)
-
-			statusBatch := make([]metrics.TxStatus, 0, len(block.Data.Data))
-			for i, data := range block.Data.Data {
-				_, channelHeader, err := serialization.UnwrapEnvelope(data)
-				if err != nil {
-					logger.Warnf("Failed to unmarshal envelope: %v", err)
-					continue
-				}
-				if common.HeaderType(channelHeader.Type) == common.HeaderType_CONFIG {
-					// We can ignore config transactions as we only count data transactions.
-					continue
-				}
-				statusBatch = append(statusBatch, metrics.TxStatus{
-					TxID:   channelHeader.TxId,
-					Status: protoblocktx.Status(statusCodes[i]),
-				})
-			}
-			processedBlocks.Write(statusBatch)
+			processedBlocks.Write(mapToStatusBatch(block))
```

Comment:
We can use multiple goroutines to process the committed status as we are unmarshaling the envelope. Otherwise, this could increase the latency unnecessarily.